include/swift/AST/NameLookup.h

```diff
@@ -536,8 +575,91 @@ class FindLocalVal : public StmtVisitor<FindLocalVal> {
   void visitCatchStmt(CatchStmt *S);
   
 };
+  
+  
+/// The bridge between the legacy UnqualifedLookupFactory and the new ASTScope
+/// lookup system
+class AbstractASTScopeDeclConsumer {
+public:
+  AbstractASTScopeDeclConsumer() {}
+
+  virtual ~AbstractASTScopeDeclConsumer() = default;
+
+  /// Called for every ValueDecl visible from the lookup.
+  /// Returns true if the lookup can be stopped at this point.
+  /// BaseDC is per legacy
+  /// Takes an array in order to batch the consumption before setting
+  /// IndexOfFirstOuterResult when necessary.
+  virtual bool consume(ArrayRef<ValueDecl *> values, DeclVisibilityKind vis,
+                       NullablePtr<DeclContext> baseDC = nullptr) = 0;
+
+  /// Eventually this functionality should move into ASTScopeLookup
+  virtual std::pair<bool, Optional<bool>>
+  lookupInSelfType(NullablePtr<DeclContext> selfDC, DeclContext *const scopeDC,
+                   NominalTypeDecl *const nominal,
+                   Optional<bool> isCascadingUse) = 0;
+
+  virtual void stopForDebuggingIfTargetLookup() = 0;
```

Comment:
Is this going to be called in the code always? If this is only for debugging the compiler, it should be hidden behind an `#ifdef`.

Reply:
Good point! Will do.